web/src/util.ts

```diff
@@ -438,6 +438,19 @@ export function get_remaining_time(start_time: number, duration: number): number
     return Math.max(0, start_time + duration - Date.now());
 }
 
+export function get_custom_time_in_minutes(time_unit: string, time_input: number): number {
+    switch (time_unit) {
+        case "hours":
+            return time_input * 60;
+        case "days":
+            return time_input * 24 * 60;
+        case "weeks":
+            return time_input * 7 * 24 * 60;
+        default:
+            return time_input;
```

Comment:
Probably it's be more robust to have a "minutes" case and throw an exception if something else was passed as the unit.

Reply:
I'm not sure if throwing an exception is correct here, so not going to change it for now.